src/main/java/ch/hevs/cloudio/endpoint/CloudioEndpoint.java

```diff
@@ -422,12 +427,18 @@ public void rollbackTransaction(){
         private int retryInterval;
         private final MqttAsyncClient mqtt;
         private final boolean persistence;
+        private int updatePersistenceLimit;
+        private int logPersistenceLimit;
         private final CloudioMessageFormat messageFormat;
         private final List<CloudioEndpointListener> listeners = new LinkedList<CloudioEndpointListener>();
         private String jobsFilePath;
         private boolean inTransaction = false;
         private Transaction transaction = new Transaction();
 
+        private Object persistenceLock = new Object();
```

Comment:
Now we could use the `cloudioPersistence` object for locking and do not need another object for that.